### ISSUE TITLE
Action::download does not work anymore

### ISSUE THREAD

**Issue by marventhieme** [laravel/nova-issues#6100]
- Laravel Version: 10.34.2
- Nova Version: 4.32.6 (works) -> 4.32.9 (does not work)
- PHP Version: 8.2.13
- Database Driver & Version: mysql 8.0
- Operating System and Version: macOS Sonoma 14.1.2
- Browser type and version: Chrome 120.0.6099.62
- Reproduction Repository: --

### Description:
I have a model "Export", which holds a filename attribute.
In a Nova Action, after creating the export model, I run:
`return Action::download(route('export.download', $export), $export->filename);`
The route goes to a controller with the following method:
```
public function __invoke(Request $request, Export $export)
    {
        return \Storage::disk('downloads')->download($export->filename);
    }
```
With Nova 4.32.6, I download the file.
With Nova 4.32.9, nothing is downloaded.
In both cases I get the notification, that the action has run successfully.
I do not get any errors.

Switching to `Action::downloadUrl(...)` does not download anything both in 4.32.6 and 4.32.9.

**Comment by mstaack** [comment]
yes, having the same. man nova seems to become more and more unstable lately...

```
  Laravel Version .......................................................................................................................... 10.35.0
  PHP Version ................................................................................................................................ 8.3.0
  Composer Version ........................................................................................................................... 2.5.5
```

```
laravel/nova                                 4.32.9             A wonderful administration interface for Laravel.
maatwebsite/laravel-nova-excel               1.3.6              Supercharged Excel exports for Laravel Nova Resources
```

**Comment by davidhemphill** [comment]
This has been fixed and will be included in the next release. 👍

**Comment by stefvanesch** [comment]
Hi @davidhemphill, any idea on when this will be released?

**Comment by mstaack** [comment]
released today. confirmed working

**Comment by simaodiascosta** [comment]
It's still broken on the latest version on both `Action::download($url, $fileName) `and `Action::downloadURL($fileName, $url)`

**Comment by glenncoppens** [comment]
Not sure if the issue I was facing, is related to this, but it might help someone else. So, the below approach might not be a fix at all, but then again, it's not really mentioned in the docs as far as I could find.
We're still using these versions. 
* Laravel Version: 9.38
* Nova 4.22.2

TLDR;
After some debugging and trying to create an equivalent of the `ExportAsCsv`-action shipped with Nova. I came to the following approach. There's a "small" note though, see further below the snippet.

```php
class ExportAction extends Action
{
    // This is used in the ActionSelector vue component to distinguish between a json/xhr response or a blobbbb.
    public $responseType = 'blob';

    // You don't necessarily need to use the `dispatchRequestUsing`-method.
    public function handle(ActionFields $fields, Collection $models)
    {
        // 1. A streamed response.
        return response()->streamDownload(function() use ($models) {
            echo 'file-content';
        });

        // 2. Or just an echo accompanied with some headers
        header('Content-Disposition: attachment; filename="file.txt";');
        echo 'file-content';
    }
}
```

Remarks on using `streamDownload`:
* if you want to provide a specific file name as a 2nd parameter to the `streamDownload` method (disposition), the file download will fail ==> I haven't looked into waht the exact issue could be.
* if you want to add specific headers you can then, eg:

1. Pass `null` as a file name and add headers.
```php
->streamDownload(..., null, ['Content-Disposition' => 'attachment; filename="file.txt";'])
```

2. Set the header just before the `echo` statement. This is basically what is being done in the `ExportAsCsv`-action.
```php
->streamDownload(function() {

    header('Content-Disposition' => 'attachment; filename="file.txt";');
    
    echo 'file-content';
})
```
